eth/tracers/api.go

```
@@ -615,25 +632,80 @@ func (api *API) traceBlock(ctx context.Context, block *types.Block, config *Trac
-		if _, err := core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas())); err != nil {
+
+		coinbaseBalance := statedb.GetBalance(blockCtx.Coinbase)
+
+		result, err := core.ApplyMessageNoFeeBurnOrTip(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()))
+
+		if err != nil {
 			failed = err
 			break
 		}
+
+		if london {
+			statedb.AddBalance(result.BurntContractAddress, result.FeeBurnt)
+		}
+
+		statedb.AddBalance(blockCtx.Coinbase, result.FeeTipped)
+		output1 := new(big.Int).SetBytes(result.SenderInitBalance.Bytes())
+		output2 := new(big.Int).SetBytes(coinbaseBalance.Bytes())
+
+		// Deprecating transfer log and will be removed in future fork. PLEASE DO NOT USE this transfer log going forward. Parameters won't get updated as expected going forward with EIP1559
+		// add transfer log
+		core.AddFeeTransferLog(
+			statedb,
+
+			msg.From(),
+			blockCtx.Coinbase,
+
+			result.FeeTipped,
+			result.SenderInitBalance,
+			coinbaseBalance,
+			output1.Sub(output1, result.FeeTipped),
+			output2.Add(output2, result.FeeTipped),
+		)
+
 		// Finalize the state so any modifications are written to the trie
 		// Only delete empty objects if EIP158/161 (a.k.a Spurious Dragon) is in effect
 		statedb.Finalise(vmenv.ChainConfig().IsEIP158(block.Number()))
+		statedb.FlushMVWriteSet()
+
+		structRead := statedb.GetReadMapDump()
+		structWrite := statedb.GetWriteMapDump()
+
+		RWstruct = append(RWstruct, structRead...)
+		RWstruct = append(RWstruct, structWrite...)
+	}
+
+	for _, val := range RWstruct {
+		IOdump += fmt.Sprintf("%v", val.TxIdx) + ", " + fmt.Sprintf("%v", val.TxInc) + ", " + fmt.Sprintf("%v", val.VerIdx) + ", " + fmt.Sprintf("%v", val.VerInc) + ", " + hex.EncodeToString(val.Path) + ", " + val.Op
 	}
+
+	// make sure that the file exists and write IOdump
+	err = ioutil.WriteFile(filepath.Join(path, "data.csv"), []byte(fmt.Sprint(IOdump)), 0600)
+	if err != nil {
+		fmt.Println("Writing IOdump to the file FAILED!\n IOdump:", IOdump)
```

Shall we return an error here?